pkg/helper/roles/helpers.go

```diff
@@ -43,6 +43,36 @@ func GetOperatorRoleName(cluster *cmv1.Cluster, missingOperator *cmv1.STSOperato
 	return awsCommonUtils.TruncateRoleName(role)
 }
 
+func CheckRedHatManagedTag(arn string, awsClient aws.Client, r *rosa.Runtime) {
+	roleName, err := aws.GetResourceIdFromARN(arn)
+	if err != nil {
+		r.Reporter.Errorf("Failed to get resource ID from ARN %s: %v", arn, err)
+		return
+	}
+
+	attachedPolicies, err := awsClient.GetPolicyDetailsFromRole(&roleName)
+	if err != nil {
+		r.Reporter.Errorf("Failed to get policy details for role '%s': %v", roleName, err)
+		return
+	}
+
+	hasRedHatManagedTag := false
```

Comment:
We don't need to get the policies in general I don't believe, related to the other comment about looping through each policy

The role itself will have the tag